### PR TITLE
feat(cli): A bit smarter cli

### PR DESCRIPTION
**Pull request by Bekacru**
Hey @shadcn and @dan5py I was seeing issues like these https://github.com/shadcn/ui/issues/818 appear repeatedly. The issue, I believe, stems from using constant initial values on the cli, so I figured it would make sense to offer better initial values based on project info. I'm not sure whether you guys are working on it with more features, but since it's not a difficult task, I thought I'd give it a shot.

### CODE REVIEW

**Comment by dan5py** [comment]
Hi @Bekacru. You're right, those kind of issues are more and more frequent. I need to test your solution but it looks promising. Thank you 💪.

**Comment by dan5py** [comment]
Just tested it out and it works very well. Just to complete your work could you please run this command:

```bash
pnpm format:write
```

Because when running the `check` it complains for the formatting. Thanks again, good job!

**Comment by dan5py** [comment]
@shadcn

**Comment by dan5py** [comment]
@shadcn

**Comment by Bekacru** [comment]
done

**Review by dan5py — APPROVED** [review]
@shadcn 👍

**Comment by shadcn** [comment]
@Bekacru Awesome work! Now that we have docs for several frameworks, I wonder if we can auto-detect framework and provide better defaults for the CLI?

**Comment by Bekacru** [comment]
@shadcn thanks! yeah that's possible. I think most of them use similar naming like app, pages, and components, as well as the folder alias is the same so we only need to tweak them slightly for each of them.  Do u want me to work on that?

**Comment by shadcn** [comment]
> Do u want me to work on that?

If you got time, yes please!

**Comment by dan5py** [comment]
I should also have time these days. I’ll see what I can do. 💪

**Comment by Bekacru** [comment]
> > Do u want me to work on that?
> 
> If you got time, yes please!

on it 👍

**Comment by Bekacru** [comment]
@shadcn and @dan5py started working on this. since all of them follow almost have similar folder structure, the present projectInfo is sufficient for the most part, but I added a styles folder resolver to better identify the global CSS. Let me know if there is anything I missed.

File: packages/cli/src/commands/init.ts [hunk 1]
```diff
@@ -27,6 +23,8 @@ import ora from "ora"
 import prompts from "prompts"
 import * as z from "zod"
 
+import { getInitialValues } from "../utils/get-initial-values"
```
Comment (dan5py) [inline packages/cli/src/commands/init.ts:26]: To respect the syntax used in the other files, use the aliased import:
```ts
import { getInitialValues } from "@/src/utils/get-initial-values"
```

File: packages/cli/src/utils/get-initial-values.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import path from "path"
+import fs from "fs-extra"
+
+import { getProjectInfo, resolveProjectDir } from "./get-project-info"
```
Comment (dan5py) [inline packages/cli/src/utils/get-initial-values.ts:4]: To respect the syntax used in the other files, use the aliased import:
```ts
import { getProjectInfo, resolveProjectDir } from "@/src/utils/get-project-info"
```

**Review by dan5py — CHANGES_REQUESTED** [review]
Thank you. I suggest just few changes.

File: packages/cli/src/utils/get-project-info.ts [hunk 3]
```diff
@@ -36,9 +39,25 @@ export async function getTsConfig() {
...
     }
-
-    return tsconfig
+    return tsconfig as TsConfigJson
   } catch (error) {
     return null
   }
 }
+
+export function getStylesDir() {
```
[the remark this answers is not in the document]
Comment (Bekacru) [reply packages/cli/src/utils/get-project-info.ts:48]: `const stylesDir = possiblePaths.find(path => existsSync(path.resolve(path))) || null;`
path will conflict with the imported path so changed it to possible path

**Comment by shadcn** [comment]
@Bekacru quick update here. This is looking good. I'll review and get this merged this week.

**Comment by shadcn** [comment]
I started testing this and it looks like we need to put in some work. We need to update the helpers so that config is read from `cwd` eg. `getTsConfig` is reading the local `tsconfig.json` and not from target dir.

**Comment by Bekacru** [comment]
> I started testing this and it looks like we need to put in some work. We need to update the helpers so that config is read from `cwd` eg. `getTsConfig` is reading the local `tsconfig.json` and not from target dir.

I'm not sure if this is what you're looking for, but adde cwd to all helper path resolvers in the same way as you did for getConfig.

**Comment by shadcn** [comment]
@Bekacru We shipped automatic detection in the cli. We're doing this on a framework-basis. I'm going to close this issue as done. Feel free to reopen if you have any questions. Thanks for your work on this PR. Appreciate it.

**Pull request closed without merging**